library/include/hipblas.h

```diff
@@ -4490,7 +4490,73 @@ HIPBLAS_EXPORT hipblasStatus_t hipblasZtrsvStridedBatched(hipblasHandle_t
 // ========== LEVEL 3 =============
 // ================================
 
-// herk
+/*! \brief BLAS Level 3 API
+
+    \details
+
+    herk performs one of the matrix-matrix operations for a Hermitian rank-k update
+
+    C := alpha*op( A )*op( A )^H + beta*C
```

Comment:
Should follow the rocblas pattern, the doc comes after the type variants for some reason @amcamd .
  We also indicate multiple matches with a convention,    xHERK would be the name referred to above following rocblas.      We are lazy but the list might even be better as you google by function name so a perfect match is best for the user:   "herk (hipblasCherk and hipbasZherk) perform one of the ...  " we want a user search to go directly to our documentation on the web.
--  @rkamd

Reply:
You're right, for rocBLAS I think the layout in readthedocs worked better if it was before the final variant. I'll make these changes.

Reply:
@TorreZuk  is correct, to produce the best output from parsing the doxygen comments they should go between the second last and last function prototype for that function. In this case, the doxygen comments would be between the hipblasCherk and hipblasZherk prototypes. You can see this in rocBLAS here: https://github.com/ROCmSoftwarePlatform/rocBLAS-internal/blob/develop/library/include/internal/rocblas-functions.h#L9702 . Where a function has s, d, c, z datatypes,  the doxygen comments would be between the second last c and the last z prototypes. This applies also for the batched and strided batched functions.  @daineAMD @rkamd 

This will also apply to @rkamd 's PR #403. Thanks for catching this Torre, I missed this in review of #403

Reply:
I changed the name to always just the blas name (herk, herkBatched, symm, symmStridedBatched, etc.). If anyone has strong feelings otherwise we can discuss.